sympy/combinatorics/rewritingsystem.py

```diff
@@ -31,11 +32,19 @@ def __init__(self, group):
         # at any point
         self._max_exceeded = False
 
+        # Reduction automaton
+        self.reduction_automaton = None
+        self._new_rules = {}
+
         # dictionary of reductions
         self.rules = {}
         self.rules_cache = deque([], 50)
         self._init_rules()
 
+        # Create a finite state machine as an instance of the StateMachine object
+        self.reduction_automaton = StateMachine('fsm')
```

Comment:
Would probably be better if the name was something like `'Reduction automaton for ' + repr(G)'` where `G` is the group.